### PR TITLE
Make netfx mean net471

### LINKED ISSUE #23165: Master should remap `netfx` to `net471`
Related #23116

We should also disable the majority of the `netfx` configurations for things inside netstandard as well as the support package since we aren't shipping them anymore.  New versions of the 4.6.1 support package will come from servicing and every library in `netstandard2.0` is now part of `net4.7.1`.

/cc @weshaggard @AlexGhiondea

### PR DESCRIPTION
**Pull request by ericstj**
This maps netfx to net471 and removes the netfx support package and corresponding
configurations from all projects.

In the cases of packages that still support netfx I only kept a live configuration in the package
if that package had type-defs.  Otherwise I just harvested the past assets.

Fixes #23165

### CODE REVIEW

**Comment by ericstj** [comment]
Note that I expect this will fail testing, I'll probably need to copy the support libs over.

**Comment by ericstj** [comment]
Yep, netfx test run is failing with
```
15:52:13 CSC : error CS0006: Metadata file 'D:\j\workspace\windows-TGrou---2a8f9c29\bin/runtime/netfx-Windows_NT-Release-x86/System.Runtime.InteropServices.RuntimeInformation.dll' could not be found [D:\j\workspace\windows-TGrou---2a8f9c29\src\System.Diagnostics.StackTrace\tests\System.Diagnostics.StackTrace.Tests.csproj]
```

Packaging build is failing because it needs my buildtools update.

**Comment by ericstj** [comment]
test OSX x64 Debug Build 
test Tizen armel Debug Build

**Comment by ericstj** [comment]
Looks like something is busted in x-plat buildtools.  Everything's failing with:
```
08:46:44 MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
08:46:44 System.IO.FileNotFoundException: Could not load file or assembly 'System.Text.Encoding.CodePages, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
```

Suspect this is related to @benaadams change: https://github.com/dotnet/buildtools/commit/1b6a978ca635aa5cbc1a9e0664d7cc4a8f10fba5

Need to investigate more.

**Comment by ericstj** [comment]
The issue is that we're using a new SDK but still have hardcoded RuntimeFrameworkVersion to [2.0.0-beta-001776-00](https://github.com/dotnet/buildtools/blame/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tool-runtime/project.csproj#L6) /cc @mellinoe 

That old version pre-dates the removal of System.Text.Encoding.Codepages, so when the tools-runtime is published it thinks Codepages is part of the shared framework and leaves it out.  We then run on the newer shared framework than was built against (due to host's roll-forward for pre-release semantics) and run on the shared framework that's missing the file.

The fix here is just to get rid RuntimeFrameworkVersion from the tool-runtime.  I suspect @mellinoe just set that because he needed a newer version than was in the CLI we were using.

**Comment by ericstj** [comment]
Ok, the latest failures also appear to be related to the updated MSBuild:
```
System.Diagnostics.Tracing.cs(122,58): error CS0067: The event 'EventListener.EventWritten' is never used [/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.csproj]
```

I repro this on windows using the CLI to build, but not when using desktop msbuild.

**Comment by ericstj** [comment]
This appears to be a breaking change in roslyn.  I'm seeing that given the same inputs *2.0.0.61404* fails, but *2.0.0.61104* succeeds.  /cc @jaredpar

**Comment by jaredpar** [comment]
> This appears to be a breaking change in roslyn. I'm seeing that given the same inputs 2.0.0.61404 fails, 

Can you point to the source location where the warning occurs? 

The only definitien I see of EventWritten is a public event in a public type. Compiler never emits unused for public members in public types. If we did then that would be such a major breaking change that we'd be drowning in GitHub issues.

**Comment by ericstj** [comment]
@safern @tarekgh it looks like the task you added is failing to run: 
```
13:24:45 D:\j\workspace\windows-TGrou---33cbf18b\Tools\tests.targets(190,5): error : Could not load file or assembly 'System.Collections.Immutable, Version=1.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. [D:\j\workspace\windows-TGrou---33cbf18b\src\tests.builds]
13:24:45 D:\j\workspace\windows-TGrou---33cbf18b\Tools\tests.targets(190,5): error :    at System.Reflection.PortableExecutable.PEReader..ctor(Stream peStream, PEStreamOptions options, Int32 size) [D:\j\workspace\windows-TGrou---33cbf18b\src\tests.builds]
13:24:45 D:\j\workspace\windows-TGrou---33cbf18b\Tools\tests.targets(190,5): error :    at Microsoft.DotNet.Build.Tasks.ExtractResWResourcesFromAssemblies.CreateReswFiles() [D:\j\workspace\windows-TGrou---33cbf18b\src\tests.builds]
13:24:45 D:\j\workspace\windows-TGrou---33cbf18b\Tools\tests.targets(190,5): error :    at Microsoft.DotNet.Build.Tasks.ExtractResWResourcesFromAssemblies.Execute() [D:\j\workspace\windows-TGrou---33cbf18b\src\tests.builds]
13:24:45 
```

I believe this is because it doesn't derive from BuildTask which does our assembly loader trick to unify dependencies: https://github.com/dotnet/buildtools/blob/03988127f489bcd1cf58d71131ac628620d53de9/src/Microsoft.DotNet.Build.Tasks.net45/BuildTask.Desktop.cs#L13

File: src/System.Runtime.InteropServices.RuntimeInformation/src/Configurations.props [hunk 1]
```diff
@@ -2,17 +2,8 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <BuildConfigurations>
-      win8-Windows_NT;
-      wpa81-Windows_NT;
-      net45-Windows_NT;
-      netfx-Windows_NT;
-      net462-Windows_NT;
-      net47-Windows_NT;
       uap-Windows_NT;
```
Comment (weshaggard) [inline src/System.Runtime.InteropServices.RuntimeInformation/src/Configurations.props]: Do we not need a new netfx configuration for this?

Comment (ericstj) [reply src/System.Runtime.InteropServices.RuntimeInformation/src/Configurations.props]: No, we no longer build that in corefx now that its part of netstandard.  It got absorbed into the framework.

Comment (weshaggard) [reply src/System.Runtime.InteropServices.RuntimeInformation/src/Configurations.props]: ah yeah :)

**Comment by ericstj** [comment]
The netfx remapping is causing many tests to fail because they are assuming 4.7/4.7.1 behavior but running on 4.6.1.

I think I'm going to tease this apart into 3 separate changes:
1. Update buildtools and fix all the breaks it introduced.
2. Remove all the netfx configurations that were statisfying the netstandard2 support package.
3. Remap netfx to net471.

File: src/System.Runtime.InteropServices.RuntimeInformation/src/Configurations.props (hunk 1 above)
Comment (tarekgh) [reply src/System.Runtime.InteropServices.RuntimeInformation/src/Configurations.props]: @ericstj will you fix the failure ``` Could not load file or assembly 'System.Collections.Immutable ``` with your changes in buildtools? or you want me or @safern take care of it?

Comment (safern) [reply src/System.Runtime.InteropServices.RuntimeInformation/src/Configurations.props]: He took care of it on: https://github.com/dotnet/buildtools/pull/1649

**Comment by ericstj** [comment]
Closing.  This was mostly subsumed by two other PRs and we're not yet doing the remapping to net471 until we have publicly available bits so that we can actually run the tests on it and have them pass.

**Pull request closed without merging**